### PR TITLE
Admission scenarios screen (EXPOSUREAPP-11751)

### PR DESCRIPTION
**Pull request by mtwalli**
## Changes 
- Admission scenarios screen 
- Update admission scenarios and selectedScenario Id after selection
- Trigger DccWalletInfo calculation
- Suppress Error per person calculation to continue for other persons
- Screenshots
- With `2.18` bug fix , UpdateTask became kind of Obsolete -> 🗑️ 
- Reaming 
```diff
-  suspend fun triggerCalculationAfterConfigChange(
+  suspend fun triggerAfterConfigChange(

- suspend fun triggerCalculationAfterCertificateChange(
+ suspend fun triggerNow(
```

## TODO
- [x] unit tests 

## Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11751

### CODE REVIEW

**Review by KathrinLuecke — DISMISSED** [review]
UA approved

**Review by chiljamgossow — DISMISSED** [review]
Looks good to, tested it
The popup seems to be dismissed right away as the calculation is fast. Maybe we can drop it if displaying it is slower than the calculation, but that would be an enhancement

**Comment by mtwalli** [comment]
> Looks good to, tested it
> The popup seems to be dismissed right away as the calculation is fast. Maybe we can drop it if displaying it is slower than the calculation, but that would be an enhancement

That is right for a few person certificates ,but in case there are many persons we need the indicator and maybe in the future Configuration will change and do more work. let us keep as in Figma

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt [hunk 1]
```diff
@@ -22,66 +21,65 @@ import javax.inject.Singleton
...
-            DefaultTaskRequest(
-                type = DccWalletInfoUpdateTask::class,
-                arguments = DccWalletInfoUpdateTask.Arguments(
-                    dccWalletInfoUpdateTriggerType = TriggeredAfterConfigUpdate(
-                        configurationChanged
-                    ),
-                    admissionScenarioId = getAdmissionScenarioId()
-                ),
-                originTag = TAG
-            )
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt:64]: Task removed

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt [hunk 2]
```diff
@@ -259,6 +260,28 @@ class PersonOverviewViewModelTest : BaseTest() {
         }
     }
 
+    @Test
+    fun `openAdmissionScenarioScreen - success`() {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt:264]: These 2 tests are fine. However, I like it a bit better if we would call `triggerNow()` in these tests and verify if `dccWalletInfoCalculationManager.triggerNow()` was called and the correct scenarioId was passed. Then we wouldn't need "@VisibleForTesting" for `admissionScenarioId()` .

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                subtitle shouldBe "Für dieses Bundesland liegen momentan keine Regeln vor"
+                enabled shouldBe false
+            }
+        }
+    }
+
+    @Test
+    fun selectScenario() {
+        instance().selectScenario("admissionScenarioId")
+        coVerifySequence {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModelTest.kt:81]: Maybe also verify here if the correct events were posted? (`Calculation` and `CalculationDone`)?

**Review by LukasLechnerDev — DISMISSED** [review]
Great work! 2 optional comments.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt (hunk 2 above)
Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt:264]: this function is not triggering DccWalletInfo ,it calculates the scenarios, but I will verify that :)

Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt:265]: updated tests

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModelTest.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModelTest.kt:82]: I tried ,but SingleLiveData with two emissions is really flaky in this test , no clue what is wrong

Comment (LukasLechnerDev) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModelTest.kt:82]: Ok. Then let's leave it as is.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt (hunk 2 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt:265]: Thanks!

**Review by LukasLechnerDev — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**